### DISCUSSION TITLE
BUG - Validations don't run when pasting an address into address1

### DISCUSSION THREAD

**Discussion by ct-Marvin** [Shopify/function-examples#467] [Feedback: Cart and Checkout Validation API]
Checkout validation functions are not triggered when a user (copy) pastes an address  into the address1 field.  

This is a problem for non blocking validations which are not retriggered when the user tries to go to the validation step or complete the purchase.

**Comment by aledalgrande** [comment]
Hey @ct-Marvin 👋🏻 

Could you provide reproduction steps? We might have a bug already logged for this, but just want to make sure it's the same. Thanks!

**Comment by ct-Marvin** [comment]
Hey @aledalgrande ,
thanks for getting back on this. Here are the reproduction steps:

1. Create a checkout validation function targeting the address1 field (for example simple regex checking if the string contains a number)
2. Important: The check needs to be "non blocking" (blockOnFailure = false)
3. Put something in a cart, go to the checkout and fill in your information (email, zip and city are necessary until the validations is fired, which is another weird behaviour to be honest. Should be triggered per field if conditions are met independent of other fields)
2. Write an address without a number somewhere else outside of the Shopify checkout (code editor, word etc.)
4. Paste it into the Address1 field
5. The check does not get triggered and the user can proceed in the checkout without the check beeing ever shown

Desired behaviour: 
1. Trigger the function either on a paste action (would be ideal)
2. Trigger the function when someone leaves the field

Bonus: Solve this bug too :sweat_smile: which refers to the point mentioned in 3 :
https://github.com/Shopify/function-examples/discussions/441

**Reply by aledalgrande** [reply]
Thanks! About 3: are you testing this with a 3-page or 1-page checkout layout?

**Reply by ct-Marvin** [reply]
@aledalgrande 1-page checkout.
If you can't reproduce the issue I'm happy to provide a loom video of that. You can shoot me a dm in the Partner Slack as "Marvin Siekmann"
